src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -164,6 +170,31 @@ entails SolverOptions{..} constraint context hints =
     forClassName ctx cn@(Qualified (Just mn) _) tys = concatMap (findDicts ctx cn) (ordNub (Nothing : Just mn : map Just (mapMaybe ctorModules tys)))
     forClassName _ _ _ = internalError "forClassName: expected qualified class name"
 
+    -- Left biased union of two row types
+    unionRows :: Type -> Type -> Type -> Maybe (Type, Type, Type, Maybe [Constraint])
+    unionRows l r o = if solverDeferErrors || not (hasKnown l || hasKnown r) then Nothing
+      else
+        let (ls, lEnd) = rowToList l
+            (rs, rEnd) = rowToList r
+            (_, oEnd) = rowToList o
+            lrUnion = ordNub (ls <> rs)
```

Comment:
Thanks! I actually tried the just unify tails but ran into the same error that this currently gets which is that you can't unify:
`(a::Int, a::Int, b::Int, b::Int)` with `(a::Int, b::Int)`
All the use cases rely on the fact that this will unify, am I wrong in assuming this?